src/browser_action/browser_action.js

```diff
@@ -90,21 +90,27 @@ function getUsers () {
 //Load event listener for "Noted" button
 function notedBtn() {
   $("#button").on("click", function(){
-    var currentUri;
+    //Get selected highlight color
+    var highlightColor = $("input[name=color]:checked").val();
+    highlightSelectedText(highlightColor);
+  });
+}
+
+
+function highlightSelectedText(highlightColor) {
+  var authResult = JSON.parse(localStorage.authResult || '{}')
+  var currentUri;
 
-    //Get current tab url
+  renderProfileView(authResult).then(() => {
+  //Get current tab url
     chrome.tabs.getSelected(null, (tab) => {
       currentUri = tab.url;
     });
 
-    //Get selected highlight color
-    var highlightColor = $("input[name=color]:checked").val();
```

Comment:
Why is `highlightColor` removed? It is used when the POST request is sent to save the note.